osu.Game/Overlays/BeatmapSet/Header.cs

```diff
@@ -68,8 +66,24 @@ private void updateDisplay()
                 downloadButtonsContainer.FadeIn(transition_duration);
                 favouriteButton.FadeIn(transition_duration);
 
-                noVideoButtons.FadeTo(BeatmapSet.OnlineInfo.HasVideo ? 0 : 1, transition_duration);
-                videoButtons.FadeTo(BeatmapSet.OnlineInfo.HasVideo ? 1 : 0, transition_duration);
+                if (BeatmapSet.OnlineInfo.HasVideo)
+                {
+                    videoButtons.Children = new[]
+                    {
+                        new DownloadButton("Download", "with Video", BeatmapSet),
```

Comment:
how about just `new DownloadButton(BeatmapSet, true/false)`  and move all text logic inside? you should be able to check the presence of video in the provided `BeatmapSet` for the case where no suffix is required.